CSharp/Library/QnAMaker/QnAMakerDialog.cs

```diff
@@ -1,110 +1,225 @@
+                            context.PrivateConversationData.SetValue("qnaMakerResults", qnaMakerResults);
+                            context.PrivateConversationData.SetValue("feedbackRecord", feedbackRecord);
+                            await this.QnAFeedbackStepAsync(context, qnaMakerResults);
+                        }
+                    }
+                    else
+                    {
+                        await context.PostAsync(qnaMakerResults.ServiceCfg.DefaultMessage);
+                        await this.DefaultWaitNextMessageAsync(context, message, qnaMakerResults);
+                    }
+                }
+                else
+                {
+                    await context.PostAsync(qnaMakerResults.ServiceCfg.DefaultMessage);
+                    await this.DefaultWaitNextMessageAsync(context, message, qnaMakerResults);
+                }
+            }
+            else
+            {
+                await context.PostAsync(qnaMakerResults.ServiceCfg.DefaultMessage);
+                await this.DefaultWaitNextMessageAsync(context, message, qnaMakerResults);
+            }
+        }
+
+        protected virtual bool IsConfidentAnswer(QnAMakerResults qnaMakerResults)
+        {
+            if (qnaMakerResults.Answers.Count < 2 || qnaMakerResults.Answers.FirstOrDefault().Score >= 99)
+            {
+                return true;
+            }
+
+            if (qnaMakerResults.Answers[0].Score - qnaMakerResults.Answers[1].Score > 20.0)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private async Task ResumeAndPostAnswer(IDialogContext context, IAwaitable<string> argument)
+        {
+            var selection = await argument;
+
+            var qnaMakerResults = default(QnAMakerResults);
+            context.PrivateConversationData.TryGetValue("qnaMakerResults", out qnaMakerResults);
+
+            if (qnaMakerResults != null)
+            {
+                bool match = false;
+                foreach (var qnaMakerResult in qnaMakerResults.Answers)
+                {
+                    if (qnaMakerResult.Questions[0].Equals(selection, StringComparison.OrdinalIgnoreCase))
+                    {
+                        context.PostAsync(qnaMakerResult.Answer);
+                        match = true;
+
+                        var feedbackRecord = default(FeedbackRecord);
+                        context.PrivateConversationData.TryGetValue("feedbackRecord", out feedbackRecord);
+
+                        if (feedbackRecord != null)
+                        {
+                            feedbackRecord.KbQuestion = qnaMakerResult.Questions.FirstOrDefault();
+                            feedbackRecord.KbAnswer = qnaMakerResult.Answer;
+
+                            var tasks =
+                                this.services.Select(
+                                    s =>
+                                    s.ActiveLearnAsync(

```

Comment:
Thanks for catching this. I missed this. Now, the static method ActiveLearnAsync is taking care of it by matching the knowledgebaseId and then continuing the call. Else it returns immediately with false.